### PR TITLE
Add srcset option to image_tag helper

### PR DESCRIPTION
**Pull request by robertomiranda**
### Summary

Adding srcset option to image_tag in order to add Responsive Images. 

**Usage:**
```ruby
image_tag("icon.png", srcset: { 'pic_640.jpg' => '640w', 'pic_1024.jpg' => '1024w', 'pic_1980.jpg' => '1980w' }, sizes: '100vw', class: 'my-image')
#=> <img src="/assets/ants_1980.jpg" srcset="/assets/pic_640.jpg 640w, /assets/pic_1024.jpg 1024w, /assets/pic_1980.jpg 1980w" sizes="100vw" class="my-image">
```

### CODE REVIEW

**Comment by matthewd** [comment]
I think we should respect `:skip_pipeline` for these images. No need to worry about controlling it separately for each -- just have them all use the value from the main `options`.

I like that this works with both a hash or an array-of-arrays; let's cover both of those in the tests.

We definitely need to allow a value that's already a string to pass through safely, too, though -- people will already be using it in that way, and we don't want to break them.

**Comment by robertomiranda** [comment]
@matthewd sure, I'll be working on that, thanks for the feedback 😁

**Comment by robertomiranda** [comment]
@matthewd I think is ready again for review, could you take a look again??

File: actionview/lib/action_view/helpers/asset_tag_helper.rb [hunk 1]
```diff
@@ -237,6 +243,13 @@ def image_tag(source, options = {})
           options[:alt] = options.fetch(:alt) { image_alt(src) }
         end
 
+        if options[:srcset] && !options[:srcset].is_a?(String)
+          options[:srcset] = options[:srcset].map do |src, size|
+            src_path = path_to_image(src, skip_pipeline: options.delete(:skip_pipeline))
```
Comment (dnl) [inline actionview/lib/action_view/helpers/asset_tag_helper.rb]: deleting the option inside the loop here seems like it won't do what you want

Comment (robertomiranda) [inline actionview/lib/action_view/helpers/asset_tag_helper.rb]: yeah you are right

**Comment by robertomiranda** [comment]
@matthewd could you take a look again over this PR??

**Pull request merged**